create_index_pages.py

```diff
@@ -21,6 +24,46 @@
   <body>
 '''
 
+HTML_PRETTY_HEADER = f'''
```

Comment:
For this file, I would probably extract all the HTML code into an HTML file and read this file in the script. This will make your script much cleaner, and you can use HTML formatters in your IDE.

Writing it directly in Python makes sense if it’s just a one-liner or something small. However, since there is quite a lot of HTML here, it’s worth extracting it.


Something like :

FILE: `header.html`:
```html
<!DOCTYPE html>
<html>
  <head>
    <meta name="pypi:repository-version" content="1.0">
    <title>Simple index</title>
  </head>
  <body>
```

FILE: `pretty_header.html:`
```html
<!DOCTYPE html>
<html>
<head>
    <meta name="pypi:repository-version" content="1.0">
    <title>Simple index</title>
    <link rel="icon" type="image/x-icon" href="{ICON}">

    <div id="head" style="height: fit-content;background-color: white;">
        <img src="{ICON}" alt="" style="width: 10%;height: 10%;">
        <h1 style="font-family: Verdana, Geneva, Tahoma, sans-serif;">Espressif's PyPI</h1>
    </div>
    <hr style="margin-bottom: 20px;">

</head>
<body>
    <style>
        a {{text-decoration: none;color: #000;}}
        div {{
            background-color: rgb(233, 233, 233);
            width: 98%;
            height: 30px;
            display: flex;
            align-items: center;
            justify-content: space-between;
            border-radius: 5px;
            padding: 5px;
            font-family: Verdana, Geneva, Tahoma, sans-serif;
        }}
        div span {{
            margin: 5px;
        }}
        br {{
            display: block;
            content: "";
            margin-top: 3px;
        }}
    </style>

```

... (similar with `footer.html`) ....

And then in python script for example:
```py
def read_html_file(file_path):
    with open(file_path, 'r') as file:
        return file.read()

HTML_HEADER = read_html_file('header.html')
HTML_PRETTY_HEADER = read_html_file('pretty_header.html').format(ICON=ICON)
HTML_FOOTER = read_html_file('footer.html')
```


(This is draft idea, if you want to use it, please review this suggestion code)

Reply:
Thank you for the review! I have updated this and added separate files for the HTML parts.

Reply:
@jakub-kocka Not sure if I'm missing something here, but the PR seems to be merged and the HTML files didn't make it to the codebase.

I can see you defined them with:
```py
HTML_HEADER = _html_loader('resources/html/header.html')
HTML_PRETTY_HEADER = _html_loader('resources/html/pretty_header.html')
HTML_FOOTER = _html_loader('resources/html/footer.html')
```
... but there are no HTML files in that path in the repo (no `resources/html` path). Maybe you forgot to commit/push them?

Reply:
The scheduled pipeline is failing with `FileNotFoundError: [Errno 2] No such file or directory: 'resources/html/header.html'`. @jakub-kocka PTAL

Reply:
Thank you, @dobairoland, @tomassebestik for pointing this out. The commit with the files got stuck in the process, sorry about that.

I have prepared [new PR](https://github.com/espressif/idf-python-wheels/pull/33) which is adding the missing resources.